pkg/restic/azure.go

```diff
@@ -67,35 +73,29 @@ func getStorageAccountKey(config map[string]string) (string, *azure.Environment,
 		return storageKey, env, nil
 	}
 
-	// 3. we need AZURE_TENANT_ID, AZURE_CLIENT_ID, AZURE_CLIENT_SECRET, AZURE_SUBSCRIPTION_ID
-	envVars, err := getRequiredValues(os.Getenv, tenantIDEnvVar, clientIDEnvVar, clientSecretEnvVar, subscriptionIDEnvVar)
-	if err != nil {
-		return "", nil, errors.Wrap(err, "unable to get all required environment variables")
-	}
-
-	// 4. check whether a different subscription ID was set for backups in config["subscriptionId"]
-	subscriptionId := envVars[subscriptionIDEnvVar]
-	if val := config[subscriptionIdConfigKey]; val != "" {
-		subscriptionId = val
+	// get subscription ID from object store config or AZURE_SUBSCRIPTION_ID environment variable
+	subscriptionID := getSubscriptionID(config)
+	if subscriptionID == "" {
+		return "", nil, errors.New("azure subscription ID not found in object store's config or in environment variable")
 	}
 
-	// 5. we need config["resourceGroup"], config["storageAccount"]
+	// we need config["resourceGroup"], config["storageAccount"]
 	if _, err := getRequiredValues(mapLookup(config), resourceGroupConfigKey, storageAccountConfigKey); err != nil {
 		return "", env, errors.Wrap(err, "unable to get all required config values")
 	}
 
-	// 6. get SPT
-	spt, err := newServicePrincipalToken(envVars[tenantIDEnvVar], envVars[clientIDEnvVar], envVars[clientSecretEnvVar], env)
+	// get authorizer
+	authorizer, err := auth.NewAuthorizerFromEnvironment()
```

Comment:
Can you please add a comment here indicating which environment variable the authorizer is getting created from?
From https://godoc.org/github.com/Azure/go-autorest/autorest/azure/auth#NewAuthorizerFromEnvironment
Looks like this can be from
> 1. Client credentials 2. Client certificate 3. Username password 4. MSI
in that order

Reply:
Am I reading that correctly?

Reply:
Yep, that's the standard Azure "credentials chain". Previously we only supported #1 from that list. Do you want me to copy/paste the godoc from that method in our code?

Reply:
That’s what i was thinking. Esp in scenario where someone reports this error. But that should be fine to investigate that API failure.

Reply:
comment updated!